.github/scripts/visualize_stopping_patterns.py

```
@@ -0,0 +1,434 @@
+            old_sym = PASS_SYMBOLS.get(old_p, "?")
+            new_sym = PASS_SYMBOLS.get(new_p, "?")
+            summaries.append(f"{sname}: {old_sym} → {new_sym}")
+
+    if summaries:
+        lines.append("変更内容: " + " | ".join(summaries))
+        lines.append("")
+
+    # Full station list in details (skip for deleted groups)
+    if group_stations:
+        num_stations = len(group_stations)
+        lines.append(f"<details>")
```

Ruff (0.15.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)